### PR TITLE
Add support python 3.11

### PR DESCRIPTION
**Pull request by juanitorduz**
Just trying and see :)

### CODE REVIEW

**Comment by juanitorduz** [comment]
Linter failing because a broken import (maybe related to https://github.com/python/cpython/pull/28618). I'll investigate!

**Comment by juanitorduz** [comment]
In https://github.com/pymc-devs/pymc-bart/pull/109/commits/ee40a66e1ffdc1b481d316bbcc25a48d0648b969 I updated `pylint` as it was causing the problem. We would have to fix some small changed to make `pylint` happy.

**Comment by juanitorduz** [comment]
A random test is failing... but does not seem not be related to the python migration.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ["3.8", "3.9", "3.10"]
+        python-version: ["3.8", "3.9", "3.10", "3.11"]
```
Comment (aloctavodia) [inline .github/workflows/test.yml:14]: Probably we can safely remove 3.8

Comment (juanitorduz) [reply .github/workflows/test.yml:14]: Alright! I'll remove it :)

Comment (juanitorduz) [reply .github/workflows/test.yml]: Removed in https://github.com/pymc-devs/pymc-bart/pull/109/commits/cc2a3409c34af64aa0a6fa23b97f82b3d3f0bdd0

**Comment by juanitorduz** [comment]
Some stochastic tests like `tests/test_pgbart.py::test_normal_sampler` sometimes fail because of randomness, shall we add a seed or reduce the precision?

**Comment by aloctavodia** [comment]
A seed should do it

**Review by aloctavodia — APPROVED** [review]

**Comment by juanitorduz** [comment]
Given https://github.com/pymc-devs/pymc-bart/pull/107, what do you think about a patch release after merging this one? :)

**Comment by aloctavodia** [comment]
Makes sense

**Pull request merged**